### PR TITLE
test(04-collective_v2.test.js): e2e test for collective

### PR DESCRIPTION
**Pull request by NotJohs**
Re [#2442](https://github.com/opencollective/opencollective/issues/2442)

### CODE REVIEW

**Comment by NotJohs** [comment]
Just testing login functionality and upload

**Comment by Betree** [comment]
@NotJohs any update on this one? Can I help?

**Comment by NotJohs** [comment]
@Betree Sorry for the delay. Got swarmed with work. Currently working on this.

**Comment by NotJohs** [comment]
@Betree I have two issues.
- The following buttons are absent from the page:
  - `edit cover` --> edit background cover image
  - `edit main color` --> edit primary color
- The budget section:
 Is there a way to dynamically add transactions to a hosted collective?
or do I have to manually create a transaction on the hosted collective?
or can I run the budget section test on the `New Collective page with euro currency`

**Comment by Betree** [comment]
> The following buttons are absent from the page:
> 
>     edit cover --> edit background cover image
>     edit main color --> edit primary color
> 

Which collective are you using? Are you properly logged in?

> Is there a way to dynamically add transactions to a hosted collective?
> or do I have to manually create a transaction on the hosted collective?

Not at the moment.

> or can I run the budget section test on the New Collective page with euro currency

Feel free to do that or to ignore this test (you bounty will still be counted as valid).

**Comment by NotJohs** [comment]
@Betree `testuser admin`

**Comment by NotJohs** [comment]
@betree Is there anyway we can enable color picker and background image for `testuser admin` account

File: test/cypress/integration/04-collective_v2.test.js [hunk 1]
```diff
@@ -1,41 +1,86 @@
...
+    it('Must have links to twitter, github and website', () => {
+      cy.get('[data-cy=twitterProfileUrl]').should('have.attr', 'href', 'https://twitter.com/testCollective');
+      cy.get('[data-cy=githubProfileUrl]').should('have.attr', 'href', 'https://github.com/testCollective');
+      cy.get('[data-cy=collectiveWebsite]').should('have.attr', 'href', 'https://opencollective.com/testCollective');
     });
 
-    it.skip('Must have the fiscal host displayed', () => {
-      // TODO
+    it('Must have the fiscal host displayed', () => {
+      cy.get('[data-cy=fiscalHostName]').should('have.visible');
```
Comment (Betree) [inline test/cypress/integration/04-collective_v2.test.js]: Let's check that it properly shows the host name, here it should be something like `Open Source Collective`

`cy.get('[data-cy=fiscalHostName]').contains(______);`

File: test/cypress/integration/04-collective_v2.test.js [hunk 2]
```diff
@@ -1,41 +1,86 @@
...
+};
 
+describe('New collective page', () => {
+  let collectiveSlug = null;
   before(() => {
-    cy.createHostedCollective()
+    cy.createHostedCollective({
+      twitterHandle: 'testCollective',
+      githubHandle: 'testCollective',
+      website: 'opencollective.com/testCollective',
```
Comment (Betree) [inline test/cypress/integration/04-collective_v2.test.js]: Great!

File: test/cypress/integration/04-collective_v2.test.js [hunk 3]
```diff
@@ -1,41 +1,86 @@
...
     });
 
-    it.skip('Can change avatar', () => {
-      // TODO
+    it('Can change avatar', () => {
+      uploadImage({
+        dropzone: '[data-cy=heroAvatarDropzone]',
+        file: 'gophercon.jpg',
+      });
+      cy.get('[data-cy=heroAvatarDropzoneSave]').click();
```
Comment (Betree) [inline test/cypress/integration/04-collective_v2.test.js]: How do we check that the image has properly been updated?

File: test/cypress/integration/04-collective_v2.test.js [hunk 4]
```diff
@@ -1,41 +1,86 @@
...
 
-    it.skip('Can change cover background image', () => {
-      // TODO
+    it('Can change cover background image', () => {
+      cy.get('[data-cy=edit-collective-display-features] [data-cy=edit-cover-btn]').click({ force: true });
+      uploadImage({
+        dropzone: '[data-cy=heroBackgroundDropzone]',
+        file: 'gopherBack.png',
+      });
+      cy.get('[data-cy=heroBackgroundDropzoneSave]').click();
```
Comment (Betree) [inline test/cypress/integration/04-collective_v2.test.js]: How do we check that the image has properly been updated?

File: test/cypress/integration/04-collective_v2.test.js (hunk 1 above)
Comment (NotJohs) [reply test/cypress/integration/04-collective_v2.test.js]: Understood

File: test/cypress/integration/04-collective_v2.test.js (hunk 3 above)
Comment (NotJohs) [reply test/cypress/integration/04-collective_v2.test.js]: I will look into this.

File: test/cypress/integration/04-collective_v2.test.js (hunk 4 above)
Comment (NotJohs) [reply test/cypress/integration/04-collective_v2.test.js]: I will look into this.

**Comment by znarf** [comment]
Thank you @NotJohs

**Comment by znarf** [comment]
Forgot to squash, my bad.

**Pull request merged**